demos/inpage/src/accounts/SafeECDSAAccountWrapper.ts

```diff
@@ -135,4 +148,114 @@ export default class SafeECDSAAccountWrapper implements IAccount {
 
     return await ownerWallet.signMessage(ethers.getBytes(userOpHash));
   }
+
+  async enableRecoveryModule(recoveryAddress: string) {
+    const provider = this.waxInPage.ethersProvider;
+    const owner = new ethers.Wallet(this.privateKey, provider);
+    const { chainId } = provider._network;
+
+    const safeProxy = Safe__factory.connect(this.address, owner);
+    const safeProxyAddress = await safeProxy.getAddress();
+
+    const viewer = new SafeSingletonFactoryViewer(owner, chainId);
+    const recoveryPluginDeployed = await viewer.isDeployed(
+      SafeECDSARecoveryPlugin__factory,
+      [safeProxyAddress, recoveryAddress],
+    );
+
+    let recoveryPlugin: SafeECDSARecoveryPlugin;
+    if (recoveryPluginDeployed) {
+      recoveryPlugin = viewer.connectAssume(SafeECDSARecoveryPlugin__factory, [
+        safeProxyAddress,
+        recoveryAddress,
+      ]);
+    } else {
+      const admin = await this.waxInPage.requestAdminAccount(
+        'deploy-contracts',
+      );
+      const factory = await SafeSingletonFactory.init(admin);
+
+      // TODO: (merge-ok) deploy recovery plugin on initial contracts deploy
+      recoveryPlugin = await factory.connectOrDeploy(
+        SafeECDSARecoveryPlugin__factory,
+        [safeProxyAddress, recoveryAddress],
+      );
+    }
+
+    const recoveryPluginAddress = await recoveryPlugin.getAddress();
+
+    let moduleEnabled = await safeProxy.isModuleEnabled(recoveryPluginAddress);
+    if (!moduleEnabled) {
+      // TODO: (merge-ok) try calling 'enableModule' via bundler
+      await executeContractCallWithSigners(
+        safeProxy,
+        safeProxy,
+        'enableModule',
+        [recoveryPluginAddress],
+        [owner],
+      );
+
+      moduleEnabled = await safeProxy.isModuleEnabled(recoveryPluginAddress);
+      assert(moduleEnabled, 'module not enabled');
+
+      this.recoveryAddress = recoveryAddress;
+      await this.waxInPage.storage.accounts.set([this.toData()]);
+    }
+  }
+
+  async recoveryAccount(
+    newOwnerAccountSeedPhrase: string,
```

Comment:
Hmm I didn't think about this.  Curious to here your thoughts about whether we should pass the seed phrase here or the address.  I like the idea of the recovery account being able to rest the address without knowing the seed phrase.  However, it might make it more confusing for the demo to enter an address since the Safe still has the same address.

Reply:
It might not be that confusing if we explain/demonstrate better that since right now we are using an ECDSAplugin, an eoa can actually control the safe.  So recovery is really resetting the owning EOA

Reply:
Yeah I originally just passed the new owner address in as that's all the contract call needs. The issue is that broke the wrapper API as it has a private key field, so there would be a mismatch between the address and private key in the demo the way things are. Agree passing the address is easier/better.

Perhaps we can do some brainstorming and jump on a call to discuss better ways of doing this?

Reply:
Regarding the confusion with the address, yeah feel like we should think about what address info we're showing the user and what makes more sense. We don't want them to give them like 4 addresses to remember but the delineation between safe address and owner sounds reasonable